network-config-analyzer/K8sPolicyYamlParser.py

```diff
@@ -290,13 +318,9 @@ def parse_policy(self):
                                  'finalizers': 0, 'generateName': 0, 'generation': 0, 'labels': 0, 'managedFields': 0,
                                  'ownerReferences': 0, 'resourceVersion': 0, 'selfLink': 0, 'uid': 0}
         self.check_keys_are_legal(policy_metadata, 'metadata', allowed_metadata_keys)
-        if len(policy_metadata['name']) > 253:
-            self.syntax_error('a metadata name must contain no more than 253 characters', self.policy)
-        pattern = r"[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*"
-        if re.fullmatch(pattern, policy_metadata['name']) is None:
-            self.syntax_error('a metadata name must consist of lower case alphanumeric characters, '
-                              '"-" or ".", and must start and end with an alphanumeric character', self.policy)
+        self.check_dns_subdomain_name('metadata name', policy_metadata['name'])
         if 'namespace' in policy_metadata and policy_metadata['namespace'] is not None:
+            self.check_dns_label_names('metadata namespace', policy_metadata['namespace'])
```

Comment:
I believe this is wrong. name of `namespace` should comply with dns subdomain name and not dns label name.
We should call `check_dns_label_names()` on fields of `matchLabels` and `matchExpressions`

Reply:
from the k8s reference (namespace) : 
`

<!--StartFragment-->
namespace string | Namespace  defines the space within which each name must be unique. An empty  namespace is equivalent to the "default" namespace, but "default" is the  canonical representation. Not all objects are required to be scoped to a  namespace - the value of this field for those objects will be empty.   **Must be a DNS_LABEL.**
-- | --


<!--EndFragment-->

`
the metadata name is a " subdomain dns label" 

matchLabels and "key" field in matchExpressions : follows the "k8s labels and selectors reference (neither dns subdomain/ dns label)" 
https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/
 
Additionally, all of them were tested on k8s live cluster (on the updated excel file (attached to the issue), I added descriptions of the bad path tests and and the k8s' error messages I got)